src/package/packageVersionDependency.ts

```diff
@@ -218,6 +221,37 @@ export class DependencyDotProducer {
     return `strict digraph G {${EOL}${dotLines.join(EOL)}${EOL}}`;
   }
 
+  private async addSelectedNodeIds(): Promise<string[]> {
+    const selectedNodes: string[] = [];
+    if (this.subscriberPackageVersionId === VERSION_BEING_BUILT) {
+      selectedNodes.push(this.subscriberPackageVersionId);
+    } else if (this.subscriberPackageVersionId.startsWith('04t')) {
+      selectedNodes.push(this.subscriberPackageVersionId);
+      const query = `SELECT Dependencies FROM SubscriberPackageVersion WHERE Id = '${this.subscriberPackageVersionId}'`;
+      try {
+        const result = await this.connection.tooling.query<{
+          Dependencies: { ids: Array<{ subscriberPackageVersionId: string }> } | null;
+        }>(query);
+        if (result.records?.length !== 1) {
+          return selectedNodes;
+        }
+        const dependencies = result.records[0].Dependencies;
+        if (!dependencies) {
+          return selectedNodes;
+        }
+        if (dependencies.ids && Array.isArray(dependencies.ids)) {
+          const dependencyIds = dependencies.ids
+            .map((dep) => dep.subscriberPackageVersionId)
+            .filter((id) => id && typeof id === 'string' && id.startsWith('04t'));
+          selectedNodes.push(...dependencyIds);
+        }
+      } catch (error) {
+        throw messages.createError('invalidPackageVersionIdError', [this.subscriberPackageVersionId]);
```

Comment:
@MimiRaiSalesforce - throwing away the original error is non recommended.  You lose the entire stack trace and context.  The user is presented with just 1 possible cause (invalid package version ID) and not a good way to fix things (retry).  What if it was due to something else?  The error would then be wrong and misleading.

Also, this is a library and not necessarily executed by a CLI so library error messages should not refer to command execution.